stl/inc/ranges

```diff
@@ -1913,6 +1913,50 @@ namespace ranges {
         };
 
         inline constexpr _Drop_fn drop;
+
+        // VARIABLE views::counted
+        class _Counted_fn {
+        private:
+            enum class _St { _Span, _Subrange, _Subrange_counted };
+
+            template <class _It>
+            _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
+                if constexpr (contiguous_iterator<_It>) {
+                    return {_St::_Span, noexcept(span{_STD to_address(_STD declval<_It>()), iter_difference_t<_It>{}})};
+                } else if constexpr (random_access_iterator<_It>) {
+                    return {_St::_Subrange,
+                        noexcept(subrange{_STD declval<_It>(), _STD declval<_It>() + iter_difference_t<_It>{}})};
+                } else {
+                    return {_St::_Subrange_counted,
+                        noexcept(subrange{
+                            counted_iterator{_STD declval<_It>(), iter_difference_t<_It>{}}, default_sentinel})};
+                }
+            }
+
+            template <class _It>
+            static constexpr _Choice_t<_St> _Choice = _Choose<_It>();
+
+        public:
+            // clang-format off
+            template <input_or_output_iterator _It, class _Diff>
+                requires convertible_to<_Diff, iter_difference_t<_It>>
+            _NODISCARD constexpr auto operator()(_It _First, const _Diff _Count) const
+                // clang-format on
+                noexcept(_Choice<_It>._No_throw) {
+                _STL_VERIFY(_Count >= 0, "The size passed to views::counted must be non-negative");
+                if constexpr (_Choice<_It>._Strategy == _St::_Span) {
+                    return span{
+                        _STD to_address(_First), static_cast<size_t>(static_cast<iter_difference_t<_It>>(_Count))};
+                } else if constexpr (_Choice<_It>._Strategy == _St::_Subrange) {
+                    return subrange{_First, _First + static_cast<iter_difference_t<_It>>(_Count)};
+                } else if constexpr (_Choice<_It>._Strategy == _St::_Subrange_counted) {
+                    return subrange{counted_iterator{_STD move(_First), static_cast<iter_difference_t<_It>>(_Count)},
+                        default_sentinel};
+                }
+            }
+        };
+
+        inline constexpr _Counted_fn counted;

```

Comment:
Could we please add this CPO to the `test_cpo` coverage in `P0896R4_ranges_range_machinery`?

Reply:
Done